kernel/kernel-api/src/main/java/io/delta/kernel/types/FieldMetadata.java

```diff
@@ -198,6 +198,11 @@ public Builder putFieldMetadataArray(String key, FieldMetadata[] value) {
             return this;
         }
 
+        public Builder fromMetadata(FieldMetadata meta) {
+            metadata.putAll(meta.metadata);
```

Comment:
yeah it should overwrite existing fields with the new fields from the passed metadata

Reply:
My point is more-so: should we clear `metadata` before adding all the elements from `meta.metadata`? so that contents are equal to only the elements from `meta.metadata`?

At the very least we need to document this behaviour.

What does delta-spark do?

Reply:
`delta-spark` does this: https://github.com/delta-io/delta/blob/2dfbcdcd5010801892b191b08b41148c98d7445e/spark/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala#L257-L260